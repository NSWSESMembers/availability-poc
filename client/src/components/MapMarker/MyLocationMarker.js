import React from 'react';
import PropTypes from 'prop-types';
import {
  Text,
  View,
} from 'react-native';
import { Marker } from 'react-native-maps';

import styles from './styles';


const ANCHOR = { x: 0.5, y: 0.5 };

const MyLocationMarker = ({ myPosition }) => {
  // const rotate = (typeof myPosition.heading === 'number' && myPosition.heading >= 0) ?
  // `${myPosition.heading}deg` : null; //disabled due to heading bug in library
  const rotate = null; // dont draw heading

  if (!myPosition || !myPosition.latitude || !myPosition.longitude) return null;


  return (
    <Marker
      anchor={ANCHOR}
      style={styles.mapMarker}
      coordinate={myPosition}
    >
      <View style={styles.myMarkercontainer}>
        <View style={styles.markerHalo} />
        {rotate &&
        <View style={[styles.heading, { transform: [{ rotate }] }]}>
          <View style={styles.headingPointer} />
        </View>
          }
        <View style={styles.marker}>
          <Text style={{ width: 0, height: 0 }}>
            {rotate}
          </Text>
        </View>
      </View>
    </Marker>
  );
};

MyLocationMarker.propTypes = {
  myPosition: PropTypes.shape({
    latitude: PropTypes.number,
    longitude: PropTypes.number,
    heading: PropTypes.number,
  }),
};
export default MyLocationMarker;
