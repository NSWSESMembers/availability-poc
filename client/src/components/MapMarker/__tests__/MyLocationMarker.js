import React from 'react';
import renderer from 'react-test-renderer';

import MyLocationMarker from '../MyLocationMarker';

const valid = {
  latitude: 1,
  longitude: 1,
  heading: 1,
};

const noHeading = {
  latitude: 1,
  longitude: 1,
};

test('renders correctly', () => {
  const tree1 = renderer.create(
    <MyLocationMarker
      myPosition={valid}
    />,
  ).toJSON();
  expect(tree1).toMatchSnapshot();

  const tree2 = renderer.create(
    <MyLocationMarker />,
  ).toJSON();
  expect(tree2).toMatchSnapshot();

  const tree3 = renderer.create(
    <MyLocationMarker
      myPosition={noHeading}
    />,
  ).toJSON();
  expect(tree3).toMatchSnapshot();
});
