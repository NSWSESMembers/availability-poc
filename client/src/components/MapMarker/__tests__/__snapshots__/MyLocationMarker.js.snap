// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<AIRMapMarker
  anchor={
    Object {
      "x": 0.5,
      "y": 0.5,
    }
  }
  coordinate={
    Object {
      "heading": 1,
      "latitude": 1,
      "longitude": 1,
    }
  }
  image={undefined}
  onPress={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "position": "absolute",
      },
      Object {
        "zIndex": 1000,
      },
    ]
  }
>
  <View
    style={
      Object {
        "height": 30,
        "width": 30,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 9,
          "height": 17,
          "left": 0,
          "margin": 6.5,
          "position": "absolute",
          "shadowColor": "black",
          "shadowOffset": Object {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 2,
          "top": 0,
          "width": 17,
        }
      }
    />
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "height": 30,
            "left": 0,
            "position": "absolute",
            "top": 0,
            "width": 30,
          },
          Object {
            "transform": Array [
              Object {
                "rotate": "1deg",
              },
            ],
          },
        ]
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "transparent",
            "borderBottomColor": "blue",
            "borderBottomWidth": 7,
            "borderLeftColor": "transparent",
            "borderLeftWidth": 5.25,
            "borderRightColor": "transparent",
            "borderRightWidth": 5.25,
            "borderStyle": "solid",
            "borderTopColor": "transparent",
            "borderTopWidth": 0,
            "height": 0,
            "width": 0,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "backgroundColor": "blue",
          "borderRadius": 8,
          "height": 15,
          "justifyContent": "center",
          "margin": 7.5,
          "width": 15,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "height": 0,
            "width": 0,
          }
        }
      >
        1deg
      </Text>
    </View>
  </View>
</AIRMapMarker>
`;

exports[`renders correctly 2`] = `null`;

exports[`renders correctly 3`] = `
<AIRMapMarker
  anchor={
    Object {
      "x": 0.5,
      "y": 0.5,
    }
  }
  coordinate={
    Object {
      "latitude": 1,
      "longitude": 1,
    }
  }
  image={undefined}
  onPress={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "position": "absolute",
      },
      Object {
        "zIndex": 1000,
      },
    ]
  }
>
  <View
    style={
      Object {
        "height": 30,
        "width": 30,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 9,
          "height": 17,
          "left": 0,
          "margin": 6.5,
          "position": "absolute",
          "shadowColor": "black",
          "shadowOffset": Object {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 2,
          "top": 0,
          "width": 17,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "blue",
          "borderRadius": 8,
          "height": 15,
          "justifyContent": "center",
          "margin": 7.5,
          "width": 15,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "height": 0,
            "width": 0,
          }
        }
      />
    </View>
  </View>
</AIRMapMarker>
`;
